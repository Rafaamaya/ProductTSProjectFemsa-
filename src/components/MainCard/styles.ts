import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  card: {
    borderRadius: 20,
    width: 286,
    height: 143,
    shadowColor: '#000000',
    shadowOffset: {width: -2, height: 4},
    shadowOpacity: 0.2,
    shadowRadius: 3,
    elevation: 9,
    justifyContent:'center',
  },
  cardHourText: {
    fontFamily:'Avenir',
    fontSize:16,
    fontWeight:'800',
    color: 'white',
    position:'absolute',
    marginLeft:19,
    paddingBottom:80,
  },
  cardTemparatureText: {
    fontFamily:'Avenir',
    fontSize:32,
    fontWeight:'800',
    color: 'white',
    textAlign:'center',
  },
});
