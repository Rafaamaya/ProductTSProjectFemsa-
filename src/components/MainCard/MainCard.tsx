import React from 'react';
import {ActivityIndicator, Text, View} from 'react-native';
import styles from './styles';

interface Props {
  title: string;
  points: number;
  backgroundColor: string;
  isLoading: boolean;
}

const MainCard = ({title, points, backgroundColor, isLoading}: Props) => {
  const formattedPoints = points
    .toString()
    .replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,');

  return (
    <View style={[styles.card, {backgroundColor: backgroundColor}]}>
      <Text style={styles.cardHourText}>{title}</Text>
      {isLoading ? (
        <ActivityIndicator color="#FFFFFF" size="large" />
      ) : (
        <Text style={styles.cardTemparatureText}>{formattedPoints} pts</Text>
      )}
    </View>
  );
};

export default MainCard;
