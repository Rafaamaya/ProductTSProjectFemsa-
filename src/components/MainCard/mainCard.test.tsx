import React from 'react';
import renderer from 'react-test-renderer';
import MainCard from './MainCard';

describe('Pruebas en <ButtonBox/>', () => {
  test('debe de hacer match con el snapchot', () => {
    const tree = renderer.create(<MainCard title={''} points={0} backgroundColor={''} isLoading={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
