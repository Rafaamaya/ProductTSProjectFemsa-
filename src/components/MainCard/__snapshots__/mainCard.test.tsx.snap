// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ButtonBox/> debe de hacer match con el snapchot 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 20,
        "elevation": 9,
        "height": 143,
        "justifyContent": "center",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
        "width": 286,
      },
      Object {
        "backgroundColor": "",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Avenir",
        "fontSize": 16,
        "fontWeight": "800",
        "marginLeft": 19,
        "paddingBottom": 80,
        "position": "absolute",
      }
    }
  >
    
  </Text>
  <ActivityIndicator
    color="#FFFFFF"
    size="large"
  />
</View>
`;
